Mariela Quintanar De la Mora A01642675

Valeria Aispuro Gómez A01642923

Alvaro Alejandro Cabañas Calvillo A01625742

Elius Ivan González Campos A01620856

Francisco Javier Romo Juárez A01643189


1. Carga los datos usando tu lector de csv o con pandas. Es recomendable hacerlo con pandas.
2. Verifica la cantidad de datos que tienes, las variables que contiene cada vector de datos e identifica el tipo de variables.

      La cantidad de datos en este csv son 10,000.

      Las variables contenidas en cada vector son del tipo Dtype object para la variable 'Género', y Dtype float para las variables 'Estatura' y 'Peso'.

In [ ]:
import pandas as pd

df = pd.read_csv('peso_estatura_genero.csv')
df.Estatura = df.Estatura*2.54
df.Peso = df.Peso/2.2
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 3 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Genero    10000 non-null  object 
 1   Estatura  10000 non-null  float64
 2   Peso      10000 non-null  float64
dtypes: float64(2), object(1)
memory usage: 234.5+ KB
None


3. Analiza las variables para saber que representa cada una y en que rangos se encuentran.
Si la descripción del problema no te lo indica, utiliza el máximo y el mínimo para encontrarlo.

In [ ]:
filtro_min = df[(df['Estatura'] < 150) & (df['Peso'] < 30)]
print(filtro_min.describe())
print(filtro_min)

         Estatura       Peso
count    1.000000   1.000000
mean   137.828359  29.409149
std           NaN        NaN
min    137.828359  29.409149
25%    137.828359  29.409149
50%    137.828359  29.409149
75%    137.828359  29.409149
max    137.828359  29.409149
      Genero    Estatura       Peso
9285  Female  137.828359  29.409149


In [ ]:
filtro_max = df[(df['Estatura'] > 200) & (df['Peso'] < 200)]
print(filtro_max.describe())
print(filtro_max)

         Estatura       Peso
count    1.000000    1.00000
mean   200.656806  122.72259
std           NaN        NaN
min    200.656806  122.72259
25%    200.656806  122.72259
50%    200.656806  122.72259
75%    200.656806  122.72259
max    200.656806  122.72259
     Genero    Estatura       Peso
2014   Male  200.656806  122.72259


En la variable 'Estatura' el rango es de 137.828359 a 200.656806
y en la variable 'Peso' el rango es de 29.409149 a 122.72259.

In [ ]:
import numpy as np

# promedio de estatura
media_e = df['Estatura'].mean()
print('La media de estaturas es', media_e)

# mediana de estatura
mediana_e = df['Estatura'].median()
print('La mediana de estaturas es', mediana_e)

# desviación estandar de estatura
std_e = df['Estatura'].std()
print('La desviación estándar de estaturas es', std_e)

print()

# promedio de peso
media_p = (df['Peso'].mean())
print('La media de pesos es', media_p)

# mediana de peso
mediana_p = (df['Peso'].median())
print('La mediana de pesos es', mediana_p)

# desviación estandar de peso
std_p = (df['Peso'].std())
print('La desviación estándar de pesos es', std_p)


# df.groupby(['Estatura']).mean()


La media de estaturas es 168.57360177735967
La mediana de estaturas es 168.4478980032
La desviación estándar de estaturas es 9.772721426820755

La media de pesos es 73.38198037883181
La mediana de pesos es 73.2786035
La desviación estándar de pesos es 14.594745003197938


In [ ]:
# Media recortada
sorted_df = df['Estatura'].sort_values()
sorted_df = sorted_df.reset_index(drop = True)
print()

p = 50
sum = 0
for i in range(p, len(sorted_df)-p):
  sum += sorted_df[i]
mean_recort = sum / (len(sorted_df)-(2*p))
print(mean_recort)


168.5684197802665


4. Basándose en la media, mediana y desviación estándar de cada variable, que conclusiones puedes entregar de los datos.

   De los datos anteriores podemos concluir que la estatura y el peso máximo pertenece a un hombre y la estatura y el peso mínimo pertenecen a una mujer.

   La media del peso y estatura nos proporciono una idea de cuál es el peso y estatura promedio en el conjunto de datos.  Utilizando el Índice de Masa Corporal (IMC) y basándonos en las medias de estatura (168.57) y peso (73.38), podemos concluir que la mayoría de los individuos en el archivo CSV se encuentran en la categoría de sobrepeso.

   La mediana del peso y estatura es similar a la media, esto quiere decir que los datos de peso y estatura están distribuidos de manera simétrica.

   La desviación estándar en estaturas es baja lo que indica que existe una mayor uniformidad en las estaturas mientra que en los pesos la desviación estandar es un poco más elevada lo indica una mayor variabilidad en los pesos.